setup.py

```diff
@@ -32,60 +32,59 @@
 ]
 
 TESTS_REQUIRE = [
-    'pytest >= 3.6.3',
-    'pytest-cov',
-    'coverage',
-    'flake8',
-    'nbval==0.9.6',  # https://github.com/computationalmodelling/nbval/issues/194
-    'pytest-mock',
+    "pytest >= 3.6.3",
+    "pytest-cov",
+    "coverage",
+    "flake8",
+    # nbval greater than 0.9.6 has a bug with semicolon
+    # https://github.com/computationalmodelling/nbval/issues/194
+    "nbval<=0.9.6",
+    "pytest-mock",
 ]
 
 INSTALL_REQUIRES = [
-    'matplotlib >= 3.0.0',
-    'numpy >= 1.17.3',
-    # pandas restricted to <2.1 until
-    # https://github.com/pandas-dev/pandas/issues/55794
-    # is resolved
-    'pandas >= 1.3.0, <2.1',
-    'statsmodels >= 0.11.1',
-    'scipy >= 1.2.0',
-    'h5py >= 2.8.0',
-    'plotly>=4.0.0',
-    'xgboost >= 1.3.3',
-    'pvlib >= 0.7.0, <0.11.0',
-    'scikit-learn >= 0.22.0',
-    'arch >= 4.11',
-    'filterpy >= 1.4.2'
+    "matplotlib >= 3.5.3",
+    "numpy >= 1.22.4",
+    "pandas >= 1.4.4",
+    "statsmodels >= 0.13.5",
+    "scipy >= 1.8.1",
+    "h5py >= 3.7.0",
+    "plotly>=4.0.0",
+    "xgboost >= 1.6.0",
+    "pvlib >= 0.11.0, <0.12.0",
+    "scikit-learn >= 1.1.3",
+    "arch >= 5.0",
+    "filterpy >= 1.4.2"
 ]
 
 EXTRAS_REQUIRE = {
-    'doc': [
-        'sphinx==4.5.0',
-        'nbsphinx==0.8.8',
-        'nbsphinx-link==1.3.0',
-        'sphinx_rtd_theme==0.5.2',
-        'ipython',
+    "doc": [
+        "sphinx==8.0.2",
```

Comment:
Good catch @cdeline! Looks like we made that change in #428, @martin-springer  any input or concerns? Otherwise I'm happy to try it and see if everything works.

Reply:
I don't think there should be any major issues with downgrading sphinx. The only thing that might happen is that some other doc requirements need to be downgraded as well. 

`.readthedocs.yml` has used python 3.7 in the past, which caused the documentation to fail on the current branch with all the updated packages. Thus, I had to update the python version for the readthedocs workflow and went straight to python 3.12 as we decided this to be our new default.